ex04.ex

```diff
@@ -55,7 +59,15 @@ defmodule Ex04 do
 
   """
 
-  def min . . . "your code"
+  def min([ h | t ]) do 
```

Comment:
# -1

I was going to say the same as I said for 4.1, and not deduct, but the problem here is different.

`min([list])` returns the smallest element in a list, so why should it return an empty list when passed an empty list?

In this case, it is "someone else's problem", where that someone is `reduce`. Let it take the head of the list as the initial value (by using `reduce/2` and not `reduce/3`) and let it worry about the error when the list is empty:

~~~ elixir
defp my_min(a,b) when a < b, do: a
defp my_min(a,b), do: b

def min(list), do: reduce(list, &my_min(&1, &2))
~~~